source/Posets/InductiveTypesfromSmallGeneratedLattice.lagda

```diff
@@ -0,0 +1,1758 @@
+
+\end{code}
+
+We now define monotone endo-maps on lattice. This is sufficient for our work
+as we are studying fixed-points.
+
+\begin{code}
+
+module Monotone-Endo-Maps {𝓤 𝓦 𝓥 : Universe} (L : Sup-Lattice 𝓤 𝓦 𝓥) where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ _is-monotone : (f : ⟨ L ⟩ → ⟨ L ⟩) → 𝓤 ⊔ 𝓦  ̇
+ f is-monotone = (x y : ⟨ L ⟩) → (x ≤ y) holds → (f x ≤ f y) holds
+
+\end{code}
+
+We now show that when one subset contains another the join of their total
+spaces are ordered as expected. This result will be quite useful throughout
+this file.
+
+\begin{code}
+
+module Subsets-Order-Joins {𝓤 𝓦 𝓥 : Universe}
+                           (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                           (A : 𝓥  ̇)
+                           (m : A → ⟨ L ⟩)
+                           where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ x ≤ y = order-of L x y
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ open Joins _≤_
+
+ joins-preserve-containment : {P : 𝓟 {𝓥} A} {Q : 𝓟 {𝓥} A}
+                            → (C : P ⊆ Q)
+                            → ((⋁ ((Σ a ꞉ A , a ∈ P) , m ∘ pr₁))
+                            ≤ (⋁ ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁))) holds
+ joins-preserve-containment {P} {Q} C =
+   (is-least-upper-bound-for L of ((Σ a ꞉ A , a ∈ P ) , m ∘ pr₁))
+    (⋁ ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁) ,
+    λ (b , b-in-P)
+      → (is-an-upper-bound-for L of ((Σ a ꞉ A , a ∈ Q ) , m ∘ pr₁))
+    (b , C b b-in-P))
+
+\end{code}
+
+We now show if a type is small and has a map to the carrier then it has a join.
+This seems like strict requirement but as we will see it occurs often when
+considering a lattice with a basis. 
+
+\begin{code}
+
+module Small-Types-have-Joins {𝓤 𝓦 𝓥 𝓣 : Universe}
+                              (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                              (T : 𝓣  ̇)
+                              (m : T → ⟨ L ⟩)
+                              (t : T is 𝓥 small)
+                              where
+ 
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-for L
+
+ small-type : 𝓥  ̇
+ small-type = pr₁ t
+
+ small-≃ : small-type ≃ T
+ small-≃ = pr₂ t
+
+ small-map : small-type → T
+ small-map = ⌜ small-≃ ⌝
+
+ is-equiv-small-map : is-equiv small-map
+ is-equiv-small-map = pr₂ small-≃
+
+ small-map-inv : T → small-type
+ small-map-inv =  ⌜ small-≃ ⌝⁻¹
+
+ has-section-small-map : has-section small-map
+ has-section-small-map = pr₁ is-equiv-small-map
+
+ is-section-small-map : is-section small-map
+ is-section-small-map = pr₂ is-equiv-small-map
+
+ section-small-map : small-map ∘ small-map-inv ∼ id
+ section-small-map = pr₂ has-section-small-map
+
+ retraction-small-map : small-map-inv ∘ small-map ∼ id
+ retraction-small-map = inverses-are-retractions' small-≃
+
+ small-type-inclusion : small-type → ⟨ L ⟩
+ small-type-inclusion = m ∘ small-map
+
+ s : ⟨ L ⟩
+ s = ⋁ (small-type , small-type-inclusion)
+
+ open Joins _≤_
+
+ is-lub-of-both : (s is-lub-of ((T , m))) holds
```

Comment:
I don't understand this name `is-lub-of-both`. We pride ourselves of descriptive names in `TypeTopology`, although we sin often. (Also we have too many brackets here.)

Reply:
It would be more accurate to say is-lub-of-T. But, when I use this later it is with arbitrary types that are equivalent to small types and it *kind of* makes more sense. I will leave this unresolved until I come up with a better name.